src/NuGet.Core/NuGet.ProjectModel/JsonUtility.cs

```diff
@@ -51,11 +51,18 @@ internal static PackageDependency ReadPackageDependency(string property, JToken
                 versionStr == null ? null : VersionRange.Parse(versionStr));
         }
 
+        internal static JProperty WritePackageDependencyWithLegacyString(PackageDependency item)
```

Comment:
Seems like a perfect unit test candidate.

Reply:
Pleae add unit tests for the 2 methods you're changing to protect us against future mishaps :)